layers/+lang/typescript/packages.el

```diff
@@ -19,6 +20,17 @@
         web-mode
         ))
 
+(unless (or (configuration-layer/layer-usedp 'javascript)
+            (configuration-layer/layer-usedp 'html))
+  (defun typescript/init-add-node-modules-path ()
+    (use-package add-node-modules-path :defer t)))
```

Comment:
That's surprising to me, I would think that configuration-layer/layer-usedp would return true for any layer that was declared and load order would be irrelevant. Otherwise there would be layer load order requirements and... I don't believe there are. plus, looking at the actual code:

```lisp
  (configuration-layer//declare-used-layers dotspacemacs-configuration-layers)
  (configuration-layer//declare-used-packages configuration-layer--used-layers)
  ;; then load the functions and finally configure the layers
  (configuration-layer//load-layers-files configuration-layer--used-layers
                                          '("funcs.el"))
```

it looks like `configuration-layer--used-layers` is defined before layer files are loaded.

@TheBB @syl20bnr could you clarify please?